### PR TITLE
Send provider operator version in client heartbeat

### PR DESCRIPTION
**Pull request by leelavg**
As a response to client heartbeat we'll send the provider operator version for the client operator to match, this should help the client to decide on it's upgradeability condition.

[RHSTOR-5205]

### CODE REVIEW

File: services/provider/proto/provider.proto [hunk 1]
```diff
@@ -161,4 +161,7 @@ message ReportStatusRequest{
     string clientOperatorVersion = 3;
 }
 
-message ReportStatusResponse{}
+message ReportStatusResponse{
+    // Desired version which provider operator wants client operator to match
+    string providerOperatorVersion = 1;
```
Comment (nb-ohad) [inline services/provider/proto/provider.proto:166]: Can we be more specific here? 
ODFVersion? OCSVersion?

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: services/provider/proto/provider.proto (hunk 1 above)
Comment (nb-ohad) [reply services/provider/proto/provider.proto:166]: Actually, I am not sure why the client needs to know anything about provider versions. This should be framed as a desired version not a provider component version.

So maybe `clientDesiredVersion` would work best

Comment (leelavg) [reply services/provider/proto/provider.proto]: done.

**Pull request merged**